jupyterhub_moss/spawner.py

```diff
@@ -239,6 +245,14 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
             # Set path to use from first environment for the current partition
             options["environment_path"] = partition_environments[0]["path"]
 
+        if options["environment_path"].endswith(".sif"):
+            # Use singularity image
+            self.batchspawner_singleuser_cmd = " ".join(
+                self.singularity_cmd
+                + [options["environment_path"], self.batchspawner_singleuser_cmd]
```

Comment:
Is `self.batchspawner_singleuser_cmd` even set here ?

Reply:
I would put `[*self.singularity_cmd, options["environment_path"], self.batchspawner_singleuser_cmd]` so that it is clear that `self.singularity_cmd` is a tuple (contrary to `self.batchspawner_singleuser_cmd`.

Reply:
Yes, there is a default value that we'd better use at other places IMO